community/cypher/frontend-3.4/src/test/scala/org/neo4j/cypher/internal/frontend/v3_4/parser/MultipleGraphClausesParsingTest.scala

```diff
@@ -17,104 +17,123 @@
+    val newClause: ast.New = ast.New(pattern)(pos)
 
-  test("SNAPSHOT GRAPH foo TO 'url'") {
-    yields(ast.Snapshot(graph("foo"), url("url")))
+    yields(ast.ConstructGraph(
+      clones = List(clone),
+      news = List(newClause))
+    )
   }
 
-  test("SNAPSHOT SOURCE GRAPH TO 'url'") {
-    yields(ast.Snapshot(ast.SourceGraphAs(None)(pos), url("url")))
+  test("CONSTRUCT NEW (:A)") {
+    val pattern = exp.Pattern(List(exp.EveryPath(exp.NodePattern(None, Seq(exp.LabelName("A")(pos)), None)(pos))))(pos)
+    val newClause: ast.New = ast.New(pattern)(pos)
+
+    yields(ast.ConstructGraph(news = List(newClause)))
   }
 
-  test("SNAPSHOT TARGET GRAPH TO 'url'") {
-    yields(ast.Snapshot(ast.TargetGraphAs(None)(pos), url("url")))
+  test("CONSTRUCT CLONE x AS y NEW (a:A) NEW (a)-[:T]->(y)") {
+    val clone: ast.Clone = ast.Clone(List(ast.AliasedReturnItem(varFor("x"), varFor("y"))(pos)))(pos)
+
+    val pattern1 = exp.Pattern(List(exp.EveryPath(exp.NodePattern(Some(varFor("a")), Seq(exp.LabelName("A")(pos)), None)(pos))))(pos)
+    val new1: ast.New = ast.New(pattern1)(pos)
+
+    val pattern2 = exp.Pattern(List(exp.EveryPath(exp.RelationshipChain(
+      exp.NodePattern(Some(varFor("a")), Seq.empty, None)(pos),
+      exp.RelationshipPattern(None, Seq(exp.RelTypeName("T")(pos)), None, None, exp.SemanticDirection.OUTGOING)(pos),
+      exp.NodePattern(Some(varFor("y")), Seq.empty, None)(pos))(pos)
+    )))(pos)
+    val new2: ast.New = ast.New(pattern2)(pos)
+
+    yields(ast.ConstructGraph(clones = List(clone), news = List(new1, new2)))
   }
 
-  test("RELOCATE GRAPH foo TO 'url'") {
-    yields(ast.Relocate(graph("foo"), url("url")))
+  test("CONSTRUCT") {
+    yields(ast.ConstructGraph())
   }
 
-  test("DELETE GRAPHS foo, bar") {
-    yields(ast.DeleteGraphs(Seq(varFor("foo"), varFor("bar"))))
+  test("CONSTRUCT ON foo.bar") {
+    yields(ast.ConstructGraph(on = List(ast.QualifiedGraphName("foo", List("bar")))))
   }
 
-  test("DELETE GRAPH foo") {
-    yields(ast.DeleteGraphs(Seq(varFor("foo"))))
+  test("CONSTRUCT ON foo.bar, baz.boz") {
+    yields(ast.ConstructGraph(on = List(
+      ast.QualifiedGraphName("foo", List("bar")),
+      ast.QualifiedGraphName("baz", List("boz"))
+    )))
   }
 
-  test("DELETE GRAPH foo, GRAPH bar") {
-    yields(ast.DeleteGraphs(Seq(varFor("foo"), varFor("bar"))))
+  test("CONSTRUCT SET a:A CREATE (b)") {
+    failsToParse
   }
 
-  private val nodePattern = exp.Pattern(List(exp.EveryPath(exp.NodePattern(None, List(), None)(pos))))(pos)
+  test("RETURN GRAPH") {
+    yields(ast.ReturnGraph(None))
+  }
 
-  private val complexPattern = exp.Pattern(List(
-    exp.NamedPatternPart(varFor("p"), exp.EveryPath(exp.NodePattern(None, List(), None)(pos)))(pos),
-    exp.NamedPatternPart(varFor("q"), exp.EveryPath(exp.NodePattern(None, List(), None)(pos)))(pos)
-  ))(pos)
+  // TODO: Parsing ambiguity; is it a graph name 'union' or no graph name and a UNION clause?
```

Comment:
Here is a TODO left. When is it gonna be addressed?

Reply:
The TODO is actually part of a much bigger item; not allowing reserved words as symbolic names. That is, to not allow clause names such as `UNION` to be a variable. Once that is done, then this may be solved. (This extends to ugly things like `MATCH (match) WITH match.p AS with RETURN with AS return`)

Would you prefer that I remove the TODO from the code?

Reply:
No, I just wanted to understand. But this is nothing your changes introduced. In that case it can stay there.